sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -671,6 +682,21 @@ class RocksDB(
   override protected def logName: String = s"${super.logName} $loggingId"
 }
 
+object RocksDB {
```

Comment:
Do the code blocks below need to be embedded within a singleton ? Would the blocks be invoked currently ?

cc - @HeartSaVioR - to confirm

Reply:
Maybe it won't be evaluated till it is referenced. @siying Could you please try adding a log to see whether the log is printed without referencing the object?

Reply:
Sure I'll add a logging and see how many times it is called, though I believe it is a singleton.

Reply:
We are not saying it will be called multiple times. We meant it is not clear whether this block is ever evaluated (executed once) or not (never executed).